File: solutions/go/space-age/1/space_age.go
package space

type Planet string

const earthYearInSeconds = 31557600

var planetMap = map[Planet]float64{
	"Earth":   1.0 * earthYearInSeconds,
	"Mercury": 0.2408467 * earthYearInSeconds,
	"Mars":    1.8808158 * earthYearInSeconds,
	"Venus":   0.61519726 * earthYearInSeconds,
	"Saturn":  29.447498 * earthYearInSeconds,
	"Uranus":  84.016846 * earthYearInSeconds,
	"Neptune": 164.79132 * earthYearInSeconds,
	"Jupiter": 11.862615 * earthYearInSeconds,
}

func Age(seconds float64, planet Planet) float64 {
	planetTimeValue, ok := planetMap[planet]

	if !ok {
		return -1
	}

	return seconds / planetTimeValue
}

File: solutions/javascript/space-age/1/space-age.js
const EARTH_YEAR = 31557600;
const PLANETS = {
  earth: 1.0 * EARTH_YEAR,
  jupiter: 11.862615 * EARTH_YEAR,
  mars: 1.8808158 * EARTH_YEAR,
  mercury: 0.2408467 * EARTH_YEAR,
  venus: 0.61519726 * EARTH_YEAR,
  saturn: 29.447498 * EARTH_YEAR,
  uranus: 84.016846 * EARTH_YEAR,
  neptune: 164.79132 * EARTH_YEAR,
};

const roundedNumber = (num) => {
  return Math.round(10 ** 2 * num) / 10 ** 2;
};

export const age = (planet, time) => {
  return roundedNumber(time / PLANETS[planet]);
};
